tests/xfstests/run.pm

```diff
@@ -180,14 +180,20 @@ sub tests_from_category {
 # Return matched exclude tests from groups in @GROUPLIST
 # return structure - hash
 # Group name start with ! will exclude in test, and expected to use to update blacklist
+# If TEST_RANGES contain generic tests, then exclude tests from generic folder, else will exclude tests from filesystem type folder
 sub exclude_grouplist {
     my %tests_list = ();
+    my $test_folder;
+    if ($TEST_RANGES =~ /generic/) {
+        $test_folder = "generic";
+    }
+    else {
+        $test_folder = $FSTYPE;
+    }
```

Comment:
Could you consider 
` my $test_folder = $TEST_RANGES =~ /generic/ ? "generic" : $FSTYPE

Reply:
Sure, thank you for suggestion